### ISSUE TITLE
Learn By Doing

### ISSUE THREAD

**Issue by agednotdead** [sneezymud/sneezymud#358]
Learn By Doing (LBD) was implemented in a very different Sneezy Time.  Simply put, it's too hard to learn spells and skills and it should be dramatically easier and faster to learn how to do things.   "Trigger honing" to learn spells and skills isn't fun.

LBD doesn't need a total revamp, the rates just need to be improved across the board.

**Comment by Durhamite2118** [comment]
Agreed.   I'm thinking that it may be worthwhile to, just as hp totals are now connected to level, that we do the same kind of exercise for the base class skills.

For example, having the base mage spells "learnedness"  increase as your level increases, with the current LBD system a mage will "level out" of  the utility of the base mage spells before you really learn them/become effective with them, therefore they don't get used.  Same thing applies to clerics, shamans, etc.

**Comment by drewhershey** [comment]
Some questions that might help define how we should change stuff, if people could answer them:

1) What, if anything, do you like about the current LBD system?
2) What, if anything, don't you like about it?
3) Are there any skills for which it makes sense that they'd take a long time to improve to the point where they succeed reliably against an enemy of your level? (Success against higher level enemies needs its own thread, as it's a completely different check)
4) Are there any specific skills you feel should hone much more quickly? If so, which ones, and why?

**Comment by wershlak** [comment]
So there could be ideas to completely change the system that require:
1) some planning and design
2) time to implement

in the meantime are there some skills that are super annoying to learn? What are your top 5?
Can we tweak a few to make it better without much effort?

**Comment by agednotdead** [comment]
I think LBD adds character.  I like that your skills are learned and you get that validation that it's learned more as you're using it. I don't hate the system.
LBD was put in to slow leveling down. Right now dbl xp is the norm, so at the very least, it needs to be adjusted for the faster xp. It doesn't take 4-5 DAYS (96-120 hours) of total playtime to 50.  
What I don't like are simple skills that are very difficult to learn. Defense has been a pain since its inception.
Seems like a lot of work to go 1 by 1 on every skill and adjust the rates etc. and would seem simple if it was a global modifier that is easily adjusted.

**Comment by agednotdead** [comment]
I'd say many if not most of the spells/skills in the "basic" disc for each class should be very easy to learn if not nearly mastered upon learning.  Shouldn't take a ton of effort.  Heal light, armor, bless, sorc globe, etc. The basics.

**Comment by drewhershey** [comment]
@wershlak Maybe we could start by just doubling the amount you increase your skill from each hone in the function that handles LBD and see how that feels to people, then go from there? Or even triple it?

**Comment by wershlak** [comment]
Not looking at the code but yeah there’s the lbd check via the skill success check but some skills don’t go through there. Like defense

**Comment by wershlak** [comment]
https://github.com/sneezymud/sneezymud/pull/364 addresses this but I don't think it goes far enough. I think there we should tweak individual skills to improve the overall gameplay.

- Any skill that maxes under 50% in the basic disc should start at 70% minimum so that it is relatively usable when you get it.
- Any skill that is a situational skill like most spells should start at 50% and learn like 5% per hone
- Any skill that is an active skill should learn like 5% per hone (kick dissect)
- Only skills that are used passively should hone at 1% per hone. Piercing, defense, dual wield. The chance to hone should be looked at. piercing is fine while defense is still too hard to hone.
-

**Comment by agednotdead** [comment]
> * Any skill that maxes under 50% in the basic disc should start at 70% minimum so that it is relatively usable when you get it.

I like that idea a lot.
